salt/templates/virt/libvirt_domain.jinja

```diff
@@ -3,22 +3,38 @@
         <vcpu>{{ cpu }}</vcpu>
         <memory unit='KiB'>{{ mem }}</memory>
         <currentMemory unit='KiB'>{{ mem }}</currentMemory>
+        {% if memtune %}
+            <memtune>
+                {% if 'hard_limit' in memtune and memtune.hard_limit %}
+                    <hard_limit unit="bytes">{{ memtune.hard_limit }}</hard_limit>
+                {% endif %}
+                {% if 'soft_limit' in memtune and memtune.soft_limit %}
+                    <soft_limit unit="bytes">{{ memtune.soft_limit }}</soft_limit>
+                {% endif %}
+                {% if 'swap_hard_limit' in memtune and memtune.swap_hard_limit %}
+                    <swap_hard_limit unit="bytes">{{ memtune.swap_hard_limit }}</swap_hard_limit>
+                {% endif %}
+                {% if 'min_guarantee' in memtune and memtune.min_guarantee %}
+                    <min_guarantee unit="bytes">{{ memtune.min_guarantee }}</min_guarantee>
+                {% endif %}
```

Comment:
may be use `{%-` for each if start and end tag to avoid adding too many blank lines in the generated XML. This is cosmetic though. See https://jinja.palletsprojects.com/en/2.11.x/templates/#whitespace-control for reference

Reply:
We could probably be able to set up `trim_blocks ` in some of the `env` config files. But I added `-` for now.